library/src/lib/localizator-editor/localization-editor-main/localization-editor-main.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, ContentChild, Input, OnInit, TemplateRef } from '@angular/core';
+import { LocalizationEditorInputDirective, LocalizationEditorLabel } from '../localization-editor.directives';
+
+/**
+ *  Component that represents main field with add-on and is always visible
+ *  ```html
+ *  <fd-localization-editor-main [label]="'EN'">
+ *       <input fd-localization-editor-input type="text" placeholder="EN">
+ *  </fd-localization-editor-main>
+ *  ```
+ */
+@Component({
+    selector: 'fd-localization-editor-main',
+    templateUrl: './localization-editor-main.component.html'
+})
+export class LocalizationEditorMainComponent implements OnInit {
+
+    /** The text for the right add-on. */
+    @Input() label: string;
+
+    /** @hidden */
+    type: string;
+
+    /** @hidden */
+    compact: boolean;
+
+    /** @hidden */
+    @ContentChild(LocalizationEditorInputDirective) input: LocalizationEditorInputDirective;
+
+    /** @hidden */
+    @ContentChild(LocalizationEditorLabel, { read: TemplateRef }) labelTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    ngOnInit() {
+        if (this.input) {
+            this.input.compact = this.compact;
+        }
+    }
+
+    /** @hidden */
+    public setProperties(compact: boolean, type: string) {
+        this.compact = compact;
+        this.type = type;
+        this.ngOnInit();
+    }
+}
```

Comment:
Also this component is very (identical) to the other item component. Any way to reuse this?

Reply:
There is way to add just a flag with `[main]` and `*ngIf`, because only one think that differentiate them is html template. I would like to avoid adding abstract classes.

Reply:
For now I extend localization-editor-item component by localization-editor-item component, to only template is changed.